src/foundry/common/documents/macro.mjs.d.ts

```diff
@@ -0,0 +1,188 @@
+// FOUNDRY_VERSION: 10.291
+
+import Document, { DocumentMetadata } from "../abstract/document.mjs";
+import * as CONST from "../constants.mjs";
+import * as documents from "./module.mjs";
+import * as fields from "../data/fields.mjs";
+
+declare global {
+  type MacroData = BaseMacro.Properties;
+}
+
+/**
+ * The Document definition for a Macro.
+ * Defines the DataSchema and common behaviors for a Macro which are shared between both client and server.
+ */
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface BaseMacro extends BaseMacro.Properties {}
+declare class BaseMacro extends Document<BaseMacro.SchemaField, BaseMacro.Metadata> {
+  /**
+   * @param data    - Initial data from which to construct the Macro
+   * @param context - Construction context options
+   */
+  constructor(data: BaseMacro.ConstructorData, context?: DocumentConstructionContext);
+
+  static override metadata: Readonly<BaseMacro.Metadata>;
+
+  static override defineSchema(): BaseMacro.Schema;
+
+  /**
+   * The default icon used for newly created Macro documents.
+   */
+  static DEFAULT_ICON: "icons/svg/dice-target.svg";
+
+  override testUserPermission(
+    user: documents.BaseUser,
+    permission: keyof typeof CONST.DOCUMENT_OWNERSHIP_LEVELS | CONST.DOCUMENT_OWNERSHIP_LEVELS,
+    {
+      exact
+    }?: {
+      /**
+       * Require the exact permission level requested?
+       * @defaultValue `false`
+       */
+      exact?: boolean;
+    }
+  ): boolean;
+
```

Comment:
@kmoschcau question for you:

Do we need to include definitions for class members prefixed with an underscore?

```js
  /* -------------------------------------------- */
  /*  Database Event Handlers                     */
  /* -------------------------------------------- */

  /** @inheritdoc */
  async _preCreate(data, options, user) {
    this.updateSource({ author: user.id });
  }
```

Reply:
For posterity: Yes we do include everything, even including JS-private (`#foo`) properties.